site/components/clientOnly/index.js

```diff
@@ -0,0 +1,29 @@
+import React, { PropTypes, Component } from 'react';
+
+/** Only render the children of this component if JavaScript is enabled and we are in the browser */
+class ClientOnly extends Component {
+
+  componentWillMount() {
+    this.setState({
+      javaScriptSupported: false,
+    });
```

Comment:
Can this not just be `state = { javaScriptSupported: false }` as a class property?